app/src/main/java/org/simple/clinic/summary/PatientSummaryScreenController.kt

```diff
@@ -271,69 +305,104 @@ class PatientSummaryScreenController @Inject constructor(
         }
   }
 
-  private fun handleBackAndDoneClicks(events: Observable<UiEvent>): Observable<UiChange> {
-    val callers = events
-        .ofType<PatientSummaryScreenCreated>()
-        .map { it.caller }
-
+  private fun showScheduleAppointmentSheet(events: Observable<UiEvent>): Observable<UiChange> {
     val patientUuids = events
         .ofType<PatientSummaryScreenCreated>()
         .map { it.patientUuid }
-        .replay()
-        .refCount()
 
-    val bloodPressureSaves = events
-        .ofType<PatientSummaryBloodPressureClosed>()
-        .startWith(PatientSummaryBloodPressureClosed(false))
-        .map { it.wasBloodPressureSaved }
+    val screenOpenedAt = events
+        .ofType<PatientSummaryScreenCreated>()
+        .map { it.screenCreatedTimestamp }
 
-    val bloodPressureSaveRestores = events
-        .ofType<PatientSummaryRestoredWithBPSaved>()
-        .map { it.wasBloodPressureSaved }
+    val patientSummaryItemChanges = events
+        .ofType<PatientSummaryItemChanged>()
+        .map { it.patientSummaryItems }
 
-    val mergedBloodPressureSaves = bloodPressureSaves.mergeWith(bloodPressureSaveRestores)
+    val hasSummaryItemChangedStream = Observables
+        .combineLatest(screenOpenedAt, patientSummaryItemChanges)
+        .map { (screenOpenedAt, patientSummaryItem) -> patientSummaryItem.hasItemChangedSince(screenOpenedAt) }
 
-    val allBpsForPatientDeleted = patientUuids
-        .switchMap(bpRepository::bloodPressureCount)
-        .map { recordedBpCount -> recordedBpCount == 0 }
+    val allBpsForPatientDeletedStream = events
+        .ofType<PatientSummaryAllBloodPressuresDeleted>()
+        .map { it.allBloodPressuresDeleted }
 
-    val shouldShowAppointmentSheet = Observables
-        .combineLatest(mergedBloodPressureSaves, allBpsForPatientDeleted) { wasBloodPressureSaved, recordedBpsAreDeleted ->
-          if (recordedBpsAreDeleted) false else wasBloodPressureSaved
+    val shouldShowScheduleAppointmentSheetOnBackClicksStream = Observables
```

Comment:
`shouldShowScheduleAppointmentSheetOnBackClicksStream` -> `showAppointmentSheetOnBackClicksStream`?

Sorry I'm just trying to reduce the words here; this name is so long 😅

Reply:
Oh and your other streams following this don't start with `should` so this becomes consistent

Reply:
This stream actually computes a `Boolean` on whether we should show the sheet or not. The actual showing of the sheet is handled by another stream, named `showScheduleAppointmentSheetOnBackClicks`.